tests/out/wgsl/access.wgsl

```diff
@@ -33,23 +33,23 @@ fn foo([[builtin(vertex_index)]] vi: u32) -> [[builtin(position)]] vec4<f32> {
 fn atomics() {
     var tmp: i32;
 
-    let value: i32 = atomicLoad(&bar.atom);
-    let _e6: i32 = atomicAdd(&bar.atom, 5);
+    let value: i32 = atomicLoad((&bar.atom));
+    let _e6: i32 = atomicAdd((&bar.atom), 5);
     tmp = _e6;
-    let _e9: i32 = atomicSub(&bar.atom, 5);
+    let _e9: i32 = atomicSub((&bar.atom), 5);
     tmp = _e9;
-    let _e12: i32 = atomicAnd(&bar.atom, 5);
+    let _e12: i32 = atomicAnd((&bar.atom), 5);
     tmp = _e12;
-    let _e15: i32 = atomicOr(&bar.atom, 5);
+    let _e15: i32 = atomicOr((&bar.atom), 5);
     tmp = _e15;
-    let _e18: i32 = atomicXor(&bar.atom, 5);
+    let _e18: i32 = atomicXor((&bar.atom), 5);
     tmp = _e18;
-    let _e21: i32 = atomicMin(&bar.atom, 5);
+    let _e21: i32 = atomicMin((&bar.atom), 5);
```

Comment:
I assume this is a separate issue that we are over-parenthesizing

Reply:
Yeah. Since we decide what operators to emit dynamically, no fixed parenthesization rule can be ideal. To get it right in general, we'd have to pass in information about the precedence of the surrounding operators, or return information about the precedence of the expression generated.

Reply:
I think there is a nice middle ground without using precedence - stuff that we do in MSL backend. Basically, when emitting an expression we are passing a flag "am I already bound by some scope?". If the flag is false, then the expression may spawn "(" ")" around it if needed.

Reply:
Okay, that sounds like basically saying, "Am I a primary expression?" If you're an assignment rhs, or a function argument, or something, then, no parens needed. Anything else, parens are needed. So it's a coarse-grained version of what I had in mind. `bool` is better than Yet Another Enum.